### ISSUE TITLE
Received SMS messages are not sent back to the MQTT server

### ISSUE THREAD

**Issue by deece** [ibnux/Android-SMS-Gateway-MQTT#5]
Incoming SMSes should be send back to the MQTT server.

I suggest having a few topics:
<device id>/send: Sending SMS via JSON, add a message ID field (string)
<device id>/status: Report back JSON with the message ID & status (success/fail)
<device id>/receive: Report back JSON with incoming phone number and message

**Comment by Evidlo** [comment]
Yes, it was unexpected when I realized received messages are supposed to be sent back via HTTP POST.  Why not just use the MQTT connection?

**Comment by Evidlo** [comment]
I made some small changes to the application which sends received SMS messages back to the `sms/received` topic (thanks cursor.ai).

Commit can be found [here](https://github.com/Evidlo/sms_llm/commit/b87c2c921ce541979c26c75b7e2b053486e6e429)

- received messages are sent to `sms/received`
- hardcoded to  subscribe to `sms/send` topic instead of its device id

If there is more interest I can possibly make a fork which includes the features described by deece above.

**Comment by ibnux** [comment]
i will try to add both of them, if HTTP setup, then send to HTTP, if none then send back to mqtt

**Comment by ibnux** [comment]
Now SMS received will be sent to MQTT

check readme, the different only json has "type" key

https://github.com/ibnux/Android-SMS-Gateway-MQTT/releases/tag/v2026-01-30
